packages/kbn-config-schema/src/internals/index.ts

```diff
@@ -7,75 +7,30 @@
  */
 
 import Joi from 'joi';
-import {
-  AnySchema,
-  JoiRoot,
-  Reference,
-  Rules,
-  SchemaLike,
-  State,
-  ValidationErrorItem,
-  ValidationOptions,
-} from 'joi';
+import type { JoiRoot, CustomHelpers } from 'joi';
 import { isPlainObject } from 'lodash';
 import { isDuration } from 'moment';
 import { Stream } from 'stream';
 import { ByteSizeValue, ensureByteSizeValue } from '../byte_size_value';
 import { ensureDuration } from '../duration';
 
-export { AnySchema, Reference, SchemaLike, ValidationErrorItem };
-
 function isMap<K, V>(o: any): o is Map<K, V> {
   return o instanceof Map;
 }
 
-const anyCustomRule: Rules = {
-  name: 'custom',
-  params: {
-    validator: Joi.func().maxArity(1).required(),
-  },
-  validate(params, value, state, options) {
-    let validationResultMessage;
-    try {
-      validationResultMessage = params.validator(value);
-    } catch (e) {
-      validationResultMessage = e.message || e;
-    }
-
-    if (typeof validationResultMessage === 'string') {
-      return this.createError(
-        'any.custom',
-        { value, message: validationResultMessage },
-        state,
-        options
-      );
-    }
-
-    return value;
-  },
-};
-
-/**
- * @internal
- */
-export const internals = Joi.extend([
-  {
-    name: 'any',
-
-    rules: [anyCustomRule],
-  },
+export const internals: JoiRoot = Joi.extend(
   {
-    name: 'boolean',
-
+    type: 'boolean',
     base: Joi.boolean(),
```

Comment:
Pretty much everything in the extension API changed, I will not explain all the changes in this file, but feel free to ask questions if you want info on some specific parts.